src/renderer/components/layout/main-layout-header.tsx

```diff
@@ -12,6 +13,7 @@ export const MainLayoutHeader = observer(({ cluster, className }: Props) => {
   return (
     <header className={cssNames("flex gaps align-center justify-space-between", className)}>
       <span className="cluster">{cluster.name}</span>
+      <NamespaceSelectFilter />
```

Comment:
I like this a lot but it has to get past the UX police :) . Maybe it could be disabled on pages where it clearly doesn't apply (like Namespaces). And the Nodes page, where if you view the node details page it currently lists all of the pods from all namespaces on the node. Although that could still cause performance issues for a cluster with a large number of namespaces and pods? May be better to be able to filter by namespaces even on the node details page? And it would be more obvious with the namespaces filter here.

Reply:
Unfortunately this is in conflict with the new (upcoming) TopBar logic. I'll block merging until we figure out how to proceed with this.